File: tests/fixtures/tiptap/tiptap-drop-handler-pos-precedence-broken/drop-handler-broken.ts
import { Plugin } from '@tiptap/pm/state';

export function createDropPlugin(onDrop: (pos: number, item: DataTransferItem) => void) {
  return new Plugin({
    props: {
      handleDrop(view, event) {
        const items = Array.from(event.dataTransfer?.items ?? []);
        const fileItem = items.find((i) => i.kind === 'file');
        if (!fileItem) return false;

        const dropPos = view.posAtCoords({ left: event.clientX, top: event.clientY });
        // Same precedence bug: dropPos?.pos ?? (0 - 1) = dropPos?.pos ?? -1
        const pos = dropPos?.pos ?? 0 - 1;

        onDrop(pos, fileItem);
        return true;
      },
    },
  });
}

File: tests/fixtures/tiptap/tiptap-drop-handler-pos-precedence-broken/drop_handler_broken.py
"""Server-side drop-position resolver ported from a ProseMirror posAtCoords callback."""


def resolve_drop_position(coords):
    # Same precedence bug as JS: pos or (0 - 1) = pos or -1, not (pos or 0) - 1
    pos = coords.get("pos") or 0 - 1
    return pos
